### PR TITLE
feat: support progress output in pull

### PR DESCRIPTION
**Pull request by qweeah**
**What this PR does / why we need it**:
This PR adds progress output for `oras pull`

### CODE REVIEW

File: cmd/oras/root/pull.go [hunk 1]
```diff
@@ -105,30 +106,77 @@ Example - Pull artifact files from an OCI layout archive 'layout.tar':
...
+	}
+	if pulledEmpty {
+		fmt.Println("Downloaded empty artifact")
+	}
+	fmt.Println("Pulled", opts.AnnotatedReference())
+	fmt.Println("Digest:", desc.Digest)
+	return nil
+}
+
+func (po *pullOptions) doPull(ctx context.Context, src oras.ReadOnlyTarget, dst oras.GraphTarget, opts oras.CopyOptions) (ocispec.Descriptor, bool, error) {
```
Comment (shizhMSFT) [inline cmd/oras/root/pull.go:148]: `doPull` should be a standalone method and should not be a method of `pullOptions`.

File: cmd/oras/root/pull.go [hunk 2]
```diff
@@ -105,30 +106,77 @@ Example - Pull artifact files from an OCI layout archive 'layout.tar':
...
+
+	var tracked track.GraphTarget
+	if po.TTY != nil {
+		tracked, err = track.NewTarget(dst, "Downloading", "Downloaded ", po.TTY)
+		if err != nil {
+			return ocispec.Descriptor{}, false, err
+		}
+		defer tracked.Close()
+		dst = tracked
 	}
```
Comment (shizhMSFT) [inline cmd/oras/root/pull.go:166]: nit: Like `getTrackedTarget`, it seems can also be refactored although it is invoked only once.

File: cmd/oras/root/pull.go [hunk 3]
```diff
@@ -156,13 +156,9 @@ func (po *pullOptions) doPull(ctx context.Context, src oras.ReadOnlyTarget, dst
...
 
 	var tracked track.GraphTarget
-	if po.TTY != nil {
-		tracked, err = track.NewTarget(dst, "Downloading", "Downloaded ", po.TTY)
-		if err != nil {
-			return ocispec.Descriptor{}, false, err
-		}
-		defer tracked.Close()
```
Comment (shizhMSFT) [inline cmd/oras/root/pull.go]: The equivalence of `defer tracked.Close()` is not reflected in the modified code. This is also not observed in `attach` and `push`.

Comment (qweeah) [reply cmd/oras/root/pull.go]: `doPush` will handle the closing for `attach` and `push`

**Review by shizhMSFT — APPROVED** [review]
LGTM

**Pull request merged**